app/controllers/iiif_controller.rb

```diff
@@ -562,23 +562,25 @@ def canvas_from_ia_page(page)
   end
 
   def canvas_from_page(page)
-    canvas = IIIF::Presentation::Canvas.new
-    canvas.label = page.title
-    canvas.width = page.base_width
-    canvas.height = page.base_height
-    canvas['@id'] = canvas_id_from_page(page)
-
-    annotation = IIIF::Presentation::Annotation.new
-    annotation.resource = iiif_create_image_resource(page)
-    annotation['on'] = canvas['@id']
-    annotation['@id'] = "#{url_for(:root)}image-service/#{page.id}"
-    canvas.images << annotation
+    unless page.base_width.nil? && page.base_height.nil?
```

Comment:
This should be an "or" rather than an "and" -- if either is nil, then don't generate a canvas.